Automattic/jetpack
# Search package: move search dashboard to package (2/2)

#### Changes proposed in this Pull Request:
This is the final PR to move search dashboard to the search package. The PR takes the chance to refactor some parts, including, renaming, changing to use redux hooks and some tiny bits. This increases the difficulty for reviewing as some files couldn't be detected as renamed but deleted/added - I'm sorry about that. I've tried to split off some PRs to be reviewed separatedly, and this is the minimal set I can do for the rest. If you have any suggestions, I'd be happy to hear them.

Other than the migration, the PR also added sucess / failed notice, and improved availability by allowing toggling by clicking labels.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Open a site with Search subscription
- Navigate to `/wp-admin/admin.php?page=jetpack-search`
- Ensure the page looks okay under different resolutions
- Ensure you could toggle the search module and instant search experience
- Ensure there's updating/success/failed notice accordingly
- Ensure JS tests pass `cd projects/packages/search && pnpm test`
- Ensure PHP tests pass `cd projects/packages/search && composer phpunit`
